base32.go

```diff
@@ -344,17 +335,29 @@ func (enc *Encoding) decode(dst, src []byte) (n int, end bool, err error) {
 // written. If src contains invalid base32 data, it will return the
 // number of bytes successfully written and CorruptInputError.
 // New line characters (\r and \n) are ignored.
-func (enc *Encoding) Decode(dst, src []byte) (n int, err error) {
-	src = bytes.Map(removeNewlinesMapper, src)
-	n, _, err = enc.decode(dst, src)
+func (enc *Encoding) Decode(dst, s []byte) (n int, err error) {
+	stripped := make([]byte, 0, len(s))
+	for i := 0; i < len(s); i++ {
+		c := s[i]
+		if c != '\r' && c != '\n' {
+			stripped = append(stripped, c)
+		}
+	}
+	n, _, err = enc.decode(dst, stripped)
 	return
 }
 
 // DecodeString returns the bytes represented by the base32 string s.
 func (enc *Encoding) DecodeString(s string) ([]byte, error) {
-	s = strings.Map(removeNewlinesMapper, s)
+	stripped := make([]byte, 0, len(s))
+	for i := 0; i < len(s); i++ {
```

Comment:
I could do that.  It will require an extra pass and seams like an overkill.  Eventually the go compiler should be smart enough to eliminate the bounds checks.  Most other optimizing compilers will in simple cases like this.

Do you really think it could make that much of a difference?  If so I will benchmark it and let you know.